src/mono/mono/mini/mini-arm64.c

```diff
@@ -3352,6 +3410,51 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 		sreg2 = ins->sreg2;
 		imm = ins->inst_imm;
 
+		char handler_kind = opcode_simd_status[ins->opcode - OP_START];
+		if (handler_kind == OPCODE_SIMD)
+		{
+			const int _t = get_type_size_macro (ins->inst_c1);
+    	const gboolean _f = is_type_float_macro (ins->inst_c1);
+    	const int _w = get_vector_size_macro (ins);
+
+			#undef SIMD_OP
+			#undef SIMD_OP_EX
+			#define SIMD_OP(reg_w, op, fmt, i8fun, i16fun, i32fun, i64fun, f32fun, f64fun) \
+				case SIMD_OP_CODE (VREG_##reg_w, (op), 0): { \
+					SIMD_OP_INTERNAL (code, reg_w, fmt, i8fun, i16fun, i32fun, i64fun, f32fun, f64fun); \
+					} break;
+			#define SIMD_OP_EX(reg_w, op, c, fmt, i8fun, i16fun, i32fun, i64fun, f32fun, f64fun) 
+
+			switch (SIMD_OP_CODE (_w, ins->opcode, 0)) {
+				#include "simd-arm64.h"
+			default:
+				g_assert_not_reached();
+				break;
+			}
+		}
+		else if (handler_kind == OPCODE_SIMD_EX)
+		{
+			const int _t = get_type_size_macro (ins->inst_c1);
+    	const gboolean _f = is_type_float_macro (ins->inst_c1);
+    	const int _w = get_vector_size_macro (ins);
+
+			#undef SIMD_OP
+			#undef SIMD_OP_EX
+			#define SIMD_OP(reg_w, op, fmt, i8fun, i16fun, i32fun, i64fun, f32fun, f64fun)
+			#define SIMD_OP_EX(reg_w, op, c, fmt, i8fun, i16fun, i32fun, i64fun, f32fun, f64fun) \
+				case SIMD_OP_CODE (VREG_##reg_w, (op), (c)): { \
+					SIMD_OP_INTERNAL (code, reg_w, fmt, i8fun, i16fun, i32fun, i64fun, f32fun, f64fun); \
+					} break;
+
+			switch (SIMD_OP_CODE (_w, ins->opcode, ins->inst_c0)) {
+				#include "simd-arm64.h"
+			default:
+				g_assert_not_reached();
+				break;
+			}
+		}
+		else
+		{
```

Comment:
```suggestion
		} else {
```

Reply:
This will mess up the indentitation of the whole function. It would be better to put a continue into the preceeding blocks.

Reply:
There is now a `goto` that skips over the huge `switch` if the operation is resolved by the table. The `switch` indentation can be left untouched this way.